### PR TITLE
Wildcard Option for Dashboards

### PR DESCRIPTION
**Pull request by adisaran64**
Resolves: #105 
Support a wildcard option which adds all monitored dashboard folders when creating a new context.

### CODE REVIEW

**Comment by safaci2000** [comment]
A few notes. 

1. There's documentation available here: https://software.es.net/gdg/docs/usage_guide/ That's generated from hugo, markdown.  The source files are available:   https://github.com/esnet/gdg/tree/master/documentation/content/docs.

Can you update the documentation with a few notes about the wildcard behavioral change. 

2. Can you update the integration_tests/dashboard_integration_test.go to try to retrieve all dashboards with the wildcard filter on and ensure the behavior is as expected.

File: apphelpers/contextHelper.go [hunk 1]
```diff
@@ -53,19 +50,53 @@ func NewContext(name string) {
...
+		foldersList := strings.Split(folderAnswer.Folders, ",")
+		if len(foldersList) > 0 && foldersList[0] != "" {
+			answers.MonitoredFolders = foldersList
+		} else {
+			answers.MonitoredFolders = []string{"General"}
+		}
 	} else {
-		answers.MonitoredFolders = []string{"General"}
+		answers.MonitoredFolders = []string{}
+		for _, currentConfig := range config.Config().Contexts() {
```
Comment (safaci2000) [inline apphelpers/contextHelper.go:90]: This isn't valid.  In this case you should just set the folder list to Empty and ensure the IgnoreFilters flag is set to true.

**Comment by safaci2000** [comment]
It looks like I had previous partially implemented this feature.  

Look at the Grafana Config file and move change the 

	IgnoreFilters      bool                `yaml:"ignore_filters"`
To be something like:
        FilterOverrides struct              `yaml:"filter_override"` {
           ignore_dashboard_filters bool  `yaml:`ignore_dashboard_filters"`
        
        }
        
        Should allow us to expand on that if we need to in the future.  
        
      Update the ExportDashboard and LIstDashboard references to that field and create/update an integration test to ensure this works correctly and related documentation.

File: apphelpers/contextHelper.go [hunk 2]
```diff
@@ -53,19 +50,53 @@ func NewContext(name string) {
...
 	err := survey.Ask(behaviorQuestions, &promptAnswers)
 	if err != nil {
 		log.Fatal("Failed to get valid answers to generate a new context")
 	}
 
 	//Set Watched Folders
-	foldersList := strings.Split(promptAnswers.Folders, ",")
-	if len(foldersList) > 0 && foldersList[0] != "" {
-		answers.MonitoredFolders = foldersList
+	if promptAnswers.Folders == "list" {
```
[the remark this answers is not in the document]
Comment (adisaran64) [reply apphelpers/contextHelper.go:68]: Will hold off on this update until we update the entire context wizard

File: documentation/content/docs/examples.md [hunk 3]
```diff
@@ -10,6 +10,12 @@ You can create new context configuration using an interactive setup.
 $ ./bin/gdg ctx new mycontext
 ```
 
+When creating a new context, you will be asked for authorization type, your default datasource and username/password, along with which folders you wish to manage under the context. You have three options:
+
+1. Default option ("General")
+2. List of folders you wish to manage
+3. Wildcard configuration (all monitored folders)
```
Comment (safaci2000) [inline documentation/content/docs/examples.md:17]: Wildcard is ALL folders, not all monitored folders.

**Comment by adisaran64** [comment]
@safaci2000 done!

File: apphelpers/contextHelper.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (safaci2000) [reply apphelpers/contextHelper.go:70]: Can you revert your changes for the context wizard? I missed that before.  They're still part of your changes as it stands.

[the remark this answers is not in the document]
Comment (adisaran64) [reply apphelpers/contextHelper.go]: I forgot about this, reverted the changes and will work on the wizard on the other issue.

**Review by safaci2000 — APPROVED** [review]

**Pull request merged**